packages/framework/client-logger/fluid-telemetry/src/test/containerTelemetryEndToEnd.spec.realsvc.ts

```diff
@@ -0,0 +1,241 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TinyliciousClient } from "@fluidframework/tinylicious-client/internal";
+import { SharedTree } from "@fluid-experimental/tree";
+import type Sinon from "sinon";
+import { spy } from "sinon";
+import { expect } from "chai";
+import { startTelemetry } from "../factory/index.js";
+import {
+	ContainerTelemetryEventNames,
+	type ContainerConnectedTelemetry,
+	type ContainerDisconnectedTelemetry,
+	type ContainerDisposedTelemetry,
+	type IFluidTelemetry,
+	type ITelemetryConsumer,
+} from "../index.js";
+
+// This test suite creates an actual IFluidContainer and confirms events are fired with the expected names during the expected events
+
+describe("container telemetry E2E", () => {
+	let schema;
+	let tinyliciousClient: TinyliciousClient;
+	let telemetryConsumerConsumeSpy: Sinon.SinonSpy;
+	let testTelemetryConsumer: ITelemetryConsumer;
+
+	// Simple test class that will be used as the telemetry consumer.
+	class TestTelemetryConsumer implements ITelemetryConsumer {
+		/**
+		 * Takes the incoming {@link IFluidTelemetry} and sends it to Azure App Insights
+		 */
+		public consume(event: IFluidTelemetry): void {
+			return;
+		}
+	}
+
+	before(() => {
+		tinyliciousClient = new TinyliciousClient({ connection: { port: 7070 } });
+		schema = {
+			initialObjects: {
+				sharedMap1: SharedTree,
+			},
+		};
+
+		testTelemetryConsumer = new TestTelemetryConsumer();
+		telemetryConsumerConsumeSpy = spy(testTelemetryConsumer, "consume");
+	});
+
+	beforeEach(() => {
+		// We need to reset the telemetry consumer spy for each test so the trackEvent calls
+		// from the last test don't bleed into the next one.
+		telemetryConsumerConsumeSpy.resetHistory();
+	});
+
+	it("IFluid container's 'connected' system event produces expected ContainerConnectedTelemetry using AppInsightsTelemetryConsumer", async () => {
+		const { container } = await tinyliciousClient.createContainer(schema);
+
+		const containerId = await container.attach();
+		startTelemetry({
+			container,
+			containerId,
+			consumers: [testTelemetryConsumer],
+		});
+
+		// We don't know exactly when the given container events will fire and we can't await specific events so we have to
+		// wrap the container.on(...) event handler within a promise that will be awaited at the end of the test.
+		const testCompletePromise = new Promise<void>((resolve, reject) => {
```

Comment:
We can simplify this pattern a bit. The `new Promise()` can resolve as soon as the connected events fires, we `await` it at the same time we create it, and the rest of the code can just come after that `await`. Something like this:

```ts
startTelemetry({ ... });

await new Promise<void>((resolve, reject) => { container.once("connected", () => { resolve()}); });

// Do validation
```

I'd also suggest using our `timeoutPromise` helper like [here](https://github.com/microsoft/FluidFramework/blob/1a92d061a4bcd0dfac4e4827f36dd1039da55a55/azure/packages/test/scenario-runner/src/DocCreatorRunner.ts#L106-L111). There is a chance that the container closes unexpectedly while we're waiting for it to emit connected, in which case the promise would never resolve and mocha would time out the test but we wouldn't know why/where it timed out (this is a pretty simple test with a single await so in this case we might be able to figure it out, but that's not always the case). Using `timeoutPromise` we can at least provide  an error message to give us more context of what were we waiting for that took longer than expected.